core/types/types.cc

```diff
@@ -1075,15 +1082,15 @@ TypePtr Types::applyTypeArguments(const GlobalState &gs, const CallLocs &locs, u
             if (validBounds) {
                 targs.emplace_back(argType);
             } else {
-                targs.emplace_back(Types::untypedUntracked());
+                targs.emplace_back(Types::untyped(Symbols::Magic_UntypedSource_applyTypeArguments()));
             }
 
             ++it;
         } else if (genericClass == Symbols::Hash() && i == 2) {
             auto tupleArgs = targs;
             targs.emplace_back(make_type<TupleType>(tupleArgs));
         } else {
-            targs.emplace_back(Types::untypedUntracked());
+            targs.emplace_back(Types::untyped(Symbols::Magic_UntypedSource_applyTypeArguments()));
```

Comment:
This is another error condition. In this case, the user didn't provide enough type args, and we already reported that error above.